exchanges/gateio/gateio_test.go

```diff
@@ -3301,10 +3301,45 @@ func getPairs(tb testing.TB, a asset.Item) currency.Pairs {
 func BenchmarkTimeInForceFromString(b *testing.B) {
 	for b.Loop() {
 		for _, tifString := range []string{gtcTIF, iocTIF, pocTIF, fokTIF} {
-			tif := timeInForceFromString(tifString)
-			if tif == order.UnknownTIF {
+			_, err := timeInForceFromString(tifString)
+			if err != nil {
 				b.Fatal(tifString)
 			}
 		}
 	}
 }
+
+func TestTimeInForceFromString(t *testing.T) {
+	t.Parallel()
+	_, err := timeInForceFromString("abcdef")
+	assert.ErrorIs(t, err, order.ErrUnsupportedTimeInForce)
+
+	for k, v := range map[string]order.TimeInForce{gtcTIF: order.GoodTillCancel, iocTIF: order.ImmediateOrCancel, pocTIF: order.PostOnly, fokTIF: order.FillOrKill} {
+		tif, err := timeInForceFromString(k)
+		assert.NoError(t, err)
+		assert.Equal(t, v, tif)
```

Comment:
🐒  Either use `t.Run` or `assert.Equalf` and `require.NoErrorf`.
```suggestion
		t.Run(k, func(t *testing.T) {
			tif, err := timeInForceFromString(k)
			require.NoError(t, err)
			assert.Equal(t, v, tif)
		})
```